src/views/DocumentationTopic.vue

```diff
@@ -207,17 +207,25 @@ export default {
       },
       parentTopicIdentifiers,
     }) => {
-      if (!parentTopicIdentifiers.length) return references[identifier];
+      // create an end-case fallback technology
+      const fallback = { title, url: $route.path };
+      // get the reference to the current page
+      const currentPageReference = references[identifier];
+      // if there are no parent topics at all, use the current reference or a fallback one
+      if (!parentTopicIdentifiers.length) return currentPageReference || fallback;
+      // try to use the first parent topic reference, if not a technology kind
       const first = references[parentTopicIdentifiers[0]];
-      if (first.kind !== 'technologies') return first;
+      if (first && first.kind !== 'technologies') return first;
 
       // if there is a top level collection that does not have a reference to
-      // itself, manufacture a minimal one using other available data
-      if (role === TopicRole.collection && !references[identifier]) {
-        return { title, url: $route.path };
+      // itself, manufacture a minimal one using other available data.
+      // This is to guard against bad data or missing references.
+      if (role === TopicRole.collection && !currentPageReference) {
+        return fallback;
       }
-
-      return references[parentTopicIdentifiers[1]] || references[identifier];
+      // use the second parent topic identifier, if there was a first,
+      // otherwise fallback the current page reference or the final fallback
+      return (first && references[parentTopicIdentifiers[1]]) || currentPageReference || fallback;
```

Comment:
Since this will now always return something, what's the experience when the last-resort `fallback` doesn't make sense? For example, what if the current page used as the fallback is not actually a technology that would have an index? Do things just crash in a different way in subsequent code or would we already handle that gracefully? (I think it might already be covered by the data fetching logic but just making sure)

Reply:
Fetching will work, but the way we match the technology might not 🤔  We do a compare against the `currentTechnology.url === indexTechnology.path` and that might backfire. I will see if I can fix that logic as well.